internal/data-importer/cmd/pg/import.go

```diff
@@ -0,0 +1,429 @@
+		uriStyle = "path"
+	}
+
+	log.Info().Msgf("Importing PostgreSQL data from %s to %s/%s", backupPath, namespace, dbName)
+
+	baseBackupName, dbDir := parseBackupPath(backupPath)
+
+	// prepare API scheme.
+	scheme := runtime.NewScheme()
+	everestv1alpha1.AddToScheme(scheme)
+	pgv2.AddToScheme(scheme)
+	corev1.AddToScheme(scheme)
+
+	// prepare k8s client.
+	k8sClient, err := client.New(config.GetConfigOrDie(), client.Options{Scheme: scheme})
+	if err != nil {
+		return err
+	}
+
+	// first we will pause reconciliation of the database cluster because
+	// we will be making changes to the PGCluster and we don't want Everest interfering.
+	if err := pauseDBReconciliation(ctx, k8sClient, dbName, namespace); err != nil {
+		return fmt.Errorf("failed to pause DB reconciliation: %w", err)
+	}
+	defer func() {
+		if unpauseErr := unpauseDBReconciliation(ctx, k8sClient, dbName, namespace); unpauseErr != nil {
+			log.Error().Err(err).Msg("Failed to unpause DB reconciliation")
+			err = errors.Join(err, unpauseErr)
+		}
+	}()
+
+	repoName, err := getRepoName(ctx, k8sClient, dbName, namespace)
+	if err != nil {
+		return fmt.Errorf("failed to get repo name: %w", err)
+	}
+
+	var (
+		pgBackRestSecretName = "data-import-" + dbName
+		restoreName          = "data-import-" + dbName
+	)
+
+	// always clean up on exit even if we fail at some point.
+	defer func() {
+		if cleanupErr := cleanup(ctx, k8sClient, namespace, restoreName, pgBackRestSecretName); cleanupErr != nil {
+			log.Error().Err(cleanupErr).Msg("Failed to clean up after PG import")
+			err = errors.Join(err, cleanupErr)
+		}
+	}()
+
+	if err := preparePGBackrestSecret(ctx, k8sClient, repoName, pgBackRestSecretName, accessKeyId,
+		secretAccessKey, dbName, namespace); err != nil {
+		return fmt.Errorf("failed to prepare PGBackrest secret: %w", err)
+	}
+
+	if err := preparePGBackrestRepo(ctx,
+		k8sClient, repoName, pgBackRestSecretName,
+		dbDir, uriStyle, cfg.Source.S3.Bucket, endpoint, region,
+		dbName, namespace); err != nil {
+		return fmt.Errorf("failed to prepare PGBackrest repo: %w", err)
+	}
+
+	if err := runPGRestoreAndWait(ctx, k8sClient, baseBackupName, restoreName, repoName, dbName, namespace); err != nil {
+		return fmt.Errorf("failed to run PG restore: %w", err)
+	}
+
+	log.Info().Msgf("Successfully imported PostgreSQL data from %s to %s/%s", backupPath, namespace, dbName)
+
+	// After restore is complete, the database users and passwords are reverted to those from the backup.
+	// This causes a mismatch between the actual database credentials and the Kubernetes User Secret.
+	// The PG operator does not automatically detect this drift, so it will not update the database passwords.
+	// To resolve this, we clear the User Secret, prompting the PG operator to regenerate new credentials
+	// and synchronize with the DB.
+	if err := resetUserSecrets(ctx, k8sClient, dbName, namespace); err != nil {
+		return fmt.Errorf("failed to reset user secrets: %w", err)
+	}
```

Comment:
Just a heads up that with this workaround, we no longer require users to provide a password at the time of DB creation